cql3/statements/create_table_statement.cc

```diff
@@ -452,6 +453,20 @@ std::optional<sstring> check_restricted_table_properties(
             break;
         }
     }
+    if (strategy && *strategy == sstables::compaction_strategy_type::time_window) {
+        std::map<sstring, sstring> options = cfprops.get_compaction_type_options();
+        sstables::time_window_compaction_strategy_options _twcs_options(options);
+        auto ttl = cfprops.get_default_time_to_live();
+        int max_windows = qp.db().get_config().twcs_max_window_count();
+
+        if (ttl > 0) {
+           int window_count = ttl / _twcs_options.get_sstable_window_size().count();
```

Comment:
You missed this. get_sstable_window_size() is in seconds, but this code doesn't "know" that. get_default_time_to_live() has no units, so we have to know that it's in seconds. To insulate the code against get_sstable_window_size_in_seconds, write

```c++
auto window_count = std::chrono::seconds(ttl) / _twcs_options.get_sstable_window_size();
```

Now if get_sstable_window_size() returns millseconds or nanoseconds, the code still works because std::chrono::duration auto-adjusts for units.

Reply:
plus, indents should be four spaces, not three.